### PR TITLE
[#171703313] SAML certs and Startup IDP metadata from ENV

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -116,7 +135,8 @@ export const appConfig: IApplicationConfig = {
   clientLoginRedirectionUrl: CLIENT_REDIRECTION_URL,
   loginPath: "/login",
   metadataPath: "/metadata",
-  sloPath: "/slo"
+  sloPath: "/slo",
+  startupIdpsMetadata: STARTUP_IDPS_METADATA
```
Comment (gunzip) [inline src/config.ts]: can you please document this var into th readme ?

File: src/app.ts [hunk 2]
```diff
@@ -170,7 +171,7 @@ export function newApp(
   // Setup routes
   //
 
-  return tryCatch(async () => {
+  return new Task(async () => {
```
Comment (gunzip) [inline src/app.ts]: we must ensure that this code never throws

File: src/app.ts [hunk 3]
```diff
@@ -221,30 +222,36 @@ export function newApp(
...
       const idpMetadataRefreshTimer = _.startIdpMetadataRefreshTimer();
       _.app.on("server:stop", () => clearInterval(idpMetadataRefreshTimer));
       return _.app;
     })
-    .getOrElseL(err => {
-      log.error("Fatal error during Express initialization: %s", err);
-      process.exit(1);
+    .map(_ => {
+      const spidStrategyOption = getSpidStrategyOption(_);
+      // If no one idp is configured the program execution will be stopped
```
Comment (gunzip) [inline src/app.ts]:
```suggestion
      // Process ends in case no IDP is configured
```

File: src/app.ts [hunk 4]
```diff
@@ -221,30 +222,36 @@ export function newApp(
...
+      if (
+        !spidStrategyOption?.idp ||
+        new StrMap(spidStrategyOption?.idp).reduce(true, () => false)
+      ) {
+        log.error(
+          "Fatal error during Express configuation. No one IDP available."
```
Comment (gunzip) [inline src/app.ts]:
```suggestion
          "Fatal error during application start. Cannot get IDPs metadata."
```

File: src/app.ts (hunk 2 above)
Comment (BurnedMarshal) [reply src/app.ts]: Are all static methods inside the Task that should never throw. I think that `tryCatch` before and `new Task` now, are only for chaining purposes with `withSpid`.

File: src/config.ts (hunk 1 above)
Comment (BurnedMarshal) [reply src/config.ts]: Fixed in 53626f2

**Pull request merged**